compiler/rustc_hir_analysis/src/hir_ty_lowering/errors.rs

```diff
@@ -1116,29 +1119,34 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
 
     pub fn report_trait_object_addition_traits_error(
         &self,
-        regular_traits: &Vec<TraitAliasExpansionInfo<'_>>,
+        regular_traits: &Vec<(ty::PolyTraitPredicate<'tcx>, SmallVec<[Span; 1]>)>,
     ) -> ErrorGuaranteed {
-        let first_trait = &regular_traits[0];
-        let additional_trait = &regular_traits[1];
+        let (&first_span, first_alias_spans) = regular_traits[0].1.split_last().unwrap();
```

Comment:
why `split_last`  here :thinking: this seems surprising

Reply:
we use the last span to point at the traits themselves, and all other preceding spans are trait alias expansions

Reply:
pls comment this